test/jest_integration/saml_auth.tests.ts

```diff
@@ -0,0 +1,278 @@
+        },
+        opensearch_security: {
+          auth: {
+            anonymous_auth_enabled: false,
+            type: 'saml',
+          },
+        },
+      },
+      {
+        // to make ignoreVersionMismatch setting work
+        // can be removed when we have corresponding ES version
+        dev: true,
+      }
+    );
+
+    console.log('Starting OpenSearchDashboards server..');
+    await root.setup();
+    await root.start();
+
+    console.log('Starting the Selenium Web Driver');
+    driver = new Builder().forBrowser('firefox').build();
+
+    await wreck.patch(
+      'https://localhost:9200/_plugins/_security/api/rolesmapping/all_access',
+      {
+        payload: [{
+          "op": "add",
+          "path": "/users",
+          "value": ["saml.jackson@example.com"]
+        }],
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    );
+    console.log("Starting to Download Flights Sample Data");
+    await wreck.post(
+      'http://localhost:5601/api/sample_data/flights',
+      {
+        payload: {},
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    );
+    console.log("Downloaded Sample Data");
+    const getConfigResponse = await wreck.get(
+      'https://localhost:9200/_plugins/_security/api/securityconfig',
+      {
+        rejectUnauthorized: false,
+        headers: {
+          authorization: ADMIN_CREDENTIALS,
+        },
+      }
+    );
+    const responseBody = (getConfigResponse.payload as Buffer).toString();
+    config = JSON.parse(responseBody).config;
+    const saml_config = {
+      http_enabled: true,
+      transport_enabled: false,
+      order: 5,
+      http_authenticator: {
+        challenge: true,
+        type: 'saml',
+        config: {
+          idp: {
+            metadata_url: 'http://localhost:7000/metadata',
+            entity_id: 'urn:example:idp',
+          },
+          sp: {
+            entity_id: 'https://localhost:9200',
+          },
+          kibana_url: 'http://localhost:5601',
+          exchange_key: '6aff3042-1327-4f3d-82f0-40a157ac4464',
+        },
+      },
+      authentication_backend: {
+        type: 'noop',
+        config: {},
+      },
+    };
+    try {
+      config.dynamic!.authc!.saml_auth_domain = saml_config;
+      await wreck.put(
+        'https://localhost:9200/_plugins/_security/api/securityconfig/config',
+        {
+          payload: config,
+          rejectUnauthorized: false,
+          headers: {
+            'Content-Type': 'application/json',
+            authorization: ADMIN_CREDENTIALS,
+          },
+        }
+      );
+    } catch (error) {
+      console.log('Got an error!!');
+    }
+    console.log('The Config Response is : ' + JSON.stringify(config));
+  });
+
+  afterAll( async () => {
```

Comment:
This happens after all the tests have run, isn't the data automatically cleared?

Reply:
no, the test data we ingest is not cleared.